src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt

```diff
@@ -244,22 +244,22 @@ class RsErrorAnnotator : Annotator, HighlightRangeExtension {
     private fun checkImpl(holder: AnnotationHolder, impl: RsImplItem) {
         val trait = impl.traitRef?.trait ?: return
         val traitName = trait.name ?: return
-
-        val canImplement = trait.functionList.associateBy { it.name }
-        val mustImplement = canImplement.filterValues { it.isAbstract }
+        
```

Comment:
Sorry, I don't understand clearly. In the beginning of `checkImpl` method if `traitRef` is null then we immediately return from function. You suggest to reuse already resolved value of `trait` somehow?